<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>EngineSettings</class>
 <widget class="QDialog" name="EngineSettings">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>377</width>
    <height>617</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Settings</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QLabel" name="label">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Update graphics time interval [s] (50Hz=0.02s):</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLineEdit" name="grap_update">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>0.02</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="label_2">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Max simulation maximum time step [s] :</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLineEdit" name="max_delta_t">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>0.01</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="label_9">
     <property name="text">
      <string>Min simulation maximum time step [s] :</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLineEdit" name="min_delta_t"/>
   </item>
   <item>
    <widget class="QLabel" name="lab">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>max iterations per time step</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLineEdit" name="max_iter_tstep">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>10</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="energy_check">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Energy conservation check [J/s]</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLineEdit" name="energy">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>1e-2</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="label_3">
     <property name="text">
      <string>K</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLineEdit" name="set_k">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>0.5</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="update_k">
     <property name="text">
      <string>Update K to conserve energy</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="label_6">
     <property name="text">
      <string>K add factor</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLineEdit" name="add_k">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>0.5</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="label_7">
     <property name="text">
      <string>max K</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLineEdit" name="max_k">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>0.5</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="label_8">
     <property name="text">
      <string>min K</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLineEdit" name="min_k">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>0.5</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>grap_update</sender>
   <signal>editingFinished()</signal>
   <receiver>EngineSettings</receiver>
   <slot>gui_to_data()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>266</x>
     <y>151</y>
    </hint>
    <hint type="destinationlabel">
     <x>188</x>
     <y>308</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>max_delta_t</sender>
   <signal>editingFinished()</signal>
   <receiver>EngineSettings</receiver>
   <slot>gui_to_data()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>188</x>
     <y>90</y>
    </hint>
    <hint type="destinationlabel">
     <x>188</x>
     <y>308</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>max_iter_tstep</sender>
   <signal>editingFinished()</signal>
   <receiver>EngineSettings</receiver>
   <slot>gui_to_data()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>188</x>
     <y>140</y>
    </hint>
    <hint type="destinationlabel">
     <x>188</x>
     <y>308</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>energy_check</sender>
   <signal>clicked(bool)</signal>
   <receiver>EngineSettings</receiver>
   <slot>gui_to_data()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>266</x>
     <y>270</y>
    </hint>
    <hint type="destinationlabel">
     <x>188</x>
     <y>308</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>set_k</sender>
   <signal>editingFinished()</signal>
   <receiver>EngineSettings</receiver>
   <slot>gui_to_data()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>188</x>
     <y>297</y>
    </hint>
    <hint type="destinationlabel">
     <x>188</x>
     <y>308</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>update_k</sender>
   <signal>clicked()</signal>
   <receiver>EngineSettings</receiver>
   <slot>gui_to_data()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>188</x>
     <y>326</y>
    </hint>
    <hint type="destinationlabel">
     <x>188</x>
     <y>308</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>add_k</sender>
   <signal>editingFinished()</signal>
   <receiver>EngineSettings</receiver>
   <slot>gui_to_data()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>188</x>
     <y>374</y>
    </hint>
    <hint type="destinationlabel">
     <x>188</x>
     <y>308</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>max_k</sender>
   <signal>editingFinished()</signal>
   <receiver>EngineSettings</receiver>
   <slot>gui_to_data()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>188</x>
     <y>424</y>
    </hint>
    <hint type="destinationlabel">
     <x>188</x>
     <y>308</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>min_k</sender>
   <signal>editingFinished()</signal>
   <receiver>EngineSettings</receiver>
   <slot>gui_to_data()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>188</x>
     <y>474</y>
    </hint>
    <hint type="destinationlabel">
     <x>188</x>
     <y>308</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>energy</sender>
   <signal>editingFinished()</signal>
   <receiver>EngineSettings</receiver>
   <slot>gui_to_data()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>188</x>
     <y>309</y>
    </hint>
    <hint type="destinationlabel">
     <x>188</x>
     <y>308</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>gui_to_data()</slot>
 </slots>
</ui>
